compiler-plugin/src/main/scala/com/trueaccord/scalapb/compiler/ProtobufGenerator.scala

```diff
@@ -382,8 +382,14 @@ class ProtobufGenerator(val params: GeneratorParams) extends DescriptorPimps {
       .add("}")
       .add("final override def serializedSize: Int = {")
       .indent
-      .add("if (__serializedSizeCachedValue == -1) { __serializedSizeCachedValue = __computeSerializedValue() }")
-      .add("__serializedSizeCachedValue")
+      .add("var read = __serializedSizeCachedValue")
```

Comment:
Got it, no need to change. Will merge in.

On Wed, Jun 1, 2016 at 6:32 AM, Arseny Tolmachev <notifications@github.com>
wrote:

> In
> compiler-plugin/src/main/scala/com/trueaccord/scalapb/compiler/ProtobufGenerator.scala
> <https://github.com/trueaccord/ScalaPB/pull/122#discussion_r65359890>:
>
> > @@ -382,8 +382,14 @@ class ProtobufGenerator(val params: GeneratorParams) extends DescriptorPimps {
> >        .add("}")
> >        .add("final override def serializedSize: Int = {")
> >        .indent
> > -      .add("if (__serializedSizeCachedValue == -1) { __serializedSizeCachedValue = __computeSerializedValue() }")
> > -      .add("__serializedSizeCachedValue")
> > +      .add("var read = __serializedSizeCachedValue")
>
> It's two field accesses vs one.
>
> JDK's toString hashCode and protobuf's ByteString use similar style:
>
> https://github.com/google/protobuf/blob/master/java/core/src/main/java/com/google/protobuf/ByteString.java#L842
>
> But I can make it back if you still want.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/trueaccord/ScalaPB/pull/122/files/b8d6a4371ba8ce91dee94067ec51feb798b00275#r65359890>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AASwrfRPDoFIScpVxSw_2Wp41EZEaqzCks5qHYoHgaJpZM4Irdnj>
> .
>



-- 
-Nadav